### PR TITLE
Now uses Discord nickname, rather than username, if set

### PR DESCRIPTION
**Pull request by PsiOmegaDelta**
Code is a little more verbose than I'd like because Go doesn't have ternary operators, nor method overloading.

Thought I'd share, if you at all wish to have it, now that I'd got it working. And do suggest potential improvements if you see them, because I feel quite lost in terms of language features available.

### CODE REVIEW

File: bot/discord.go [hunk 1]
```diff
@@ -199,3 +200,21 @@ func dOutgoing(nick, channel, message string) {
 
 	dSession.ChannelMessageSend(chanID, fmt.Sprintf("**<%s>** %s", nick, message))
 }
+
+func getDisplayNameForMember(member *discord.Member) string {
+	if member.Nick != "" {
```
Comment (GinjaNinja32) [inline bot/discord.go]: Perhaps change the config struct to:
```go
type DiscordConfig struct {
 	Token        string "token"
    UseNicknames bool   "use_nicknames"
}
```

and change this condition to `conf.Discord.UseNicknames && member.Nick != ""`?

I don't plan to enable this for the bridge I'm running, but I'm happy to merge this if it's a config option.

File: bot/discord.go [hunk 2]
```diff
@@ -115,26 +116,26 @@ func dMessageCreate(s *discord.Session, m *discord.MessageCreate) {
...
+			incomingDiscord("[SYSTEM]", channel, fmt.Sprintf("full message from %s: %s", iAddAntiPing(authorName), url))
 		} else {
 			for _, line := range lines {
-				incomingDiscord(m.Author.Username, channel, line)
+				incomingDiscord(authorName, channel, line)
 			}
 		}
 
-		//incomingDiscord(m.Author.Username, channel, message)
+		//incomingDiscord(authorName, channel, message)
```
Comment (GinjaNinja32) [inline bot/discord.go]: Someone used sed :)

File: bot/discord.go [hunk 3]
```diff
@@ -199,3 +200,21 @@ func dOutgoing(nick, channel, message string) {
...
+func getDisplayNameForMember(member *discord.Member) string {
+	if member.Nick != "" {
+		return member.Nick
+	}
+
+	return member.User.Username
+}
+
+func getDisplayNameForUser(user *discord.User, members []*discord.Member) string {
+	for _, m := range members {
```
Comment (GinjaNinja32) [inline bot/discord.go]: This loop should probably short-circuit out if `conf.Discord.UseNicknames` is false.

File: bot/discord.go (hunk 2 above)
Comment (PsiOmegaDelta) [reply bot/discord.go]: I saw no reason to leave the comment untouched and becoming increasingly outdated. Either I update it or I remove it.

Comment (GinjaNinja32) [reply bot/discord.go]: It's the old pre-multiline-support code, please do remove it.

**Comment by PsiOmegaDelta** [comment]
Seems my config additions broke it.
Guess I'll be debugging for a moment.

**Comment by PsiOmegaDelta** [comment]
Tweaked and appears to work at an initial glance.

**Pull request merged**